modules/navier_stokes/test/tests/finite_volume/wcns/enthalpy_equation/1d_test_h_fp-nonlinearFV.i

```diff
@@ -0,0 +1,135 @@
+L = 30
+nx = 600
+bulk_u = 0.01
+p_ref = 101325.0
+T_in = 860.
+q_source = 20000.
+advected_interp_method = 'upwind'
+
+[Mesh]
+  [gmg]
+    type = GeneratedMeshGenerator
+    dim = 1
+    xmin = 0
+    xmax = ${L}
+    nx = ${nx}
+  []
+  allow_renumbering = false
+[]
+
+[Physics]
+  [NavierStokes]
+    [Flow]
+      [flow]
+        compressibility = 'weakly-compressible'
+
+        velocity_variable = 'vel_x'
+
+        density = 'rho'
+        dynamic_viscosity = 'mu'
+
+        initial_velocity = '${bulk_u} 0 0'
+        initial_pressure = '${p_ref}'
+
+        inlet_boundaries = 'left'
+        # momentum_inlet_types = 'fixed-velocity'
+        # momentum_inlet_functors = '${bulk_u} 0'
+        momentum_inlet_types = 'flux-velocity'
+        flux_inlet_pps = '${bulk_u}'
+
+        outlet_boundaries = 'right'
+        momentum_outlet_types = 'fixed-pressure'
+        pressure_functors = '${p_ref}'
+
+        momentum_advection_interpolation = ${advected_interp_method}
+        pressure_two_term_bc_expansion = false
+        momentum_two_term_bc_expansion = false
+      []
+    []
+    [FluidHeatTransfer]
+      [energy]
+        coupled_flow_physics = flow
+
+        solve_for_enthalpy = true
+        # There are no fluid temperature (auxiliary) variable when solving for enthalpy
+        # with nonlinear finite volume, because we need T_from_p_h to be computed on-the-fly
+        # rather than on an auxiliary kernel update which does not preserve automatic differentiation
+        # fluid_temperature_variable = 'T_fluid'
+
+        fp = 'lead'
+        thermal_conductivity = 'k'
+        specific_heat = 'cp'
+
+        initial_enthalpy = '${fparse 800 * 240}'
+
+        energy_inlet_types = 'flux-velocity'
+        # specifies inlet temperature, not inlet enthalpy
+        energy_inlet_functors = '${T_in}'
```

Comment:
it troubles me that temperature is used as an energy

Reply:
it is odd but not sure how to do better with a single "energy_inlet_functors" parameter that maps to a single "energy_inlet_boundaries" parameter. I really dont want to specify an inlet because that's too much work to obtain. i think my average workflow if I had to use enthalpy would be to pull up the source code for the fluid properties and compute the enthalpy I want by hand

We would need to add maybe "energy_temperature_boundaries", but here the flux boundary conditions being used are not temperature BCs. I think the code could guess that the temperature is a "flux-inlet" with a fixed temperature from seeing the boundary appear in the two parameters.